lib/cocoapods/target/build_settings.rb

```diff
@@ -204,6 +204,11 @@ def save_as(path)
         '${PODS_BUILD_DIR}/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)'
       end
 
+      # @return [String]
+      define_build_settings_method :use_recursive_script_inputs_in_script_phases, :build_setting => true do
```

Comment:
I think this can be only under AggregateSettings class in here. i.e there is no need for xcconfig files from PodTargets to  have this flag. I guess it could be harmless and one could have custom script phases though...

Reply:
Are you saying there are no cases in which CP has script phases in Pod targets?

Reply:
not currently. Again unless there is a custom user `script_phase` specified in the `podspec`, in which case users will have to remember to enable `USE_RECURSIVE_SCRIPT_INPUTS_IN_SCRIPT_PHASES` so maybe its better what we have now..........

Reply:
It doesn't hurt anyone anyway - it's a sensible default everywhere and Apple is just being overly cautious.

Reply:
yeah I think we can keep it, its harmless since most pods do not specify custom user script phases.